src/service_info.rs

```diff
@@ -201,6 +210,21 @@ impl ServiceInfo {
         self.requires_probe = enable;
     }
 
+    /// Set whether the service is restricted to link-local addresses.
+    ///
+    /// By default, it is false.
+    pub fn set_link_local_only(&mut self, is_link_local_only: bool) {
+        self.is_link_local_only = is_link_local_only;
+    }
+
+    /// Set the supported interfaces for this service.
+    ///
+    /// Only ips associated with the provided interfaces will be published.  If set to None (default),
+    /// all interfaces are supported
+    pub fn set_interfaces(&mut self, intfs: Option<Vec<String>>) {
```

Comment:
I'm wondering if we should use `Vec<IfKind>` for `intfs` parameter.  (`IfKind` is an existing public type). It's more flexible, can remove the need of `Option`, but also more complex.  Your thoughts?